pkg/controller/v1beta1/inferenceservice/reconcilers/knative/ksvc_reconciler.go

```diff
@@ -86,13 +86,22 @@ func createKnativeService(componentMeta metav1.ObjectMeta,
 		annotations[autoscaling.ClassAnnotationKey] = autoscaling.KPA
 	}
 	lastRolledoutRevision := componentStatus.LatestRolledoutRevision
-	// This is to handle case when the latest ready revision is rolled out with 100% and then rolled back
-	// so here we need to get the revision that is previously rolled out with 100%
-	if componentStatus.LatestRolledoutRevision == componentStatus.LatestReadyRevision &&
-		componentStatus.LatestReadyRevision == componentStatus.LatestCreatedRevision &&
-		componentExtension.CanaryTrafficPercent != nil && *componentExtension.CanaryTrafficPercent < 100 {
+
+	// This is to handle case when the first revision rolled out.
+	// And the canary traffic percent of the first revision should always be 100.
+	if componentStatus.PreviousRolledoutRevision == "" {
```

Comment:
I still do not quite understand the condition here, can you elaborate more?  Initially `PreviousRolledoutRevision` is always empty and is only populated when you rolled out a second revision.  Seems the logic here should be
```golang
if componentStatus.LastestRolledoutRevision == "" && componentExtension.CanaryTrafficPercent  != nil {
     fullCanaryTrafficPercent := int64(100)
     componentExtension.CanaryTrafficPercent = &fullCanaryTrafficPercent
}
```

Reply:
Right. When isvc rollout for the first time, there is no `PreviousRolledoutRevision`. At this time, the `PreviousRolledoutRevision` should always be empty and the `CanaryTrafficPercent` should be set to 100 as well，because setting to less than 100 makes no sense at all.  If not, it will lead to the situation I discussed above.

> This can not totally remove the condition for the time being. Because in the scenario of roll out **the first revision** with traffic percent 1%, when `LatestRolledoutRevision` is not empty that means **revision 01** is ready. In this case the if condition should be hint, otherwise the `PreviousRolledoutRevision` would be the same as `LatestRolledoutRevision` instead of empty. As a result, the state of the isvc directly jumps repeatedly between true and false.
> 
> All in all, the if condition still needs to be retained.

Reply:
I think here we should check `LatestRolledoutRevision` instead of `PreviousRolledoutRevision` as `PreviousRolledoutRevision == ""` can be in the rollout of second revision rollout where  `CanaryTrafficPercent` can be allowed.

Reply:
Would check `LatestCreatedRevision` be better? As the second revision is rolled out, `LatestRolledoutRevision` may not have been updated.

Reply:
hmm though it is also possible `PreviousRolledoutRevision` is still empty when you rollout second revision where `CanaryTrafficPercent` should be allowed. `PreviousRolledoutRevision` is only updated when you fully rollout the second revision.

Reply:
I think checking `LatestRolledoutRevision` would be fine, we can document or return an error that you can only apply `CanaryTrafficPercent` until `LatestRolledoutRevision` is updated.

Reply:
If check `LatestRolledoutRevision`, when I roll out revision 02.
```
if [condition] {
     fullCanaryTrafficPercent := int64(100)
     componentExtension.CanaryTrafficPercent = &fullCanaryTrafficPercent
}
``` 
The if condition will be hit, it result in the direct full roll out of the revision 02. Is that so, or do I have a problem understanding?

Reply:
`LatestCreatedRevision` has been updated to 02 and `LatestRolledoutRevision` is 01, the if condition hit and the traffic percent will be 100.

Reply:
Maybe you are right, I need to test it again.